core/src/upgrade.rs

```diff
@@ -134,6 +135,28 @@ impl<T: AsRef<[u8]>> ProtocolName for T {
     }
 }
 
+/// Wrapper for printing a [`ProtocolName`] that is expected to be mostly ASCII
+///
+/// ```rust
+/// # use libp2p_core::upgrade::DisplayProtocolName;
+/// assert_eq!(DisplayProtocolName("hello").to_string(), "hello");
```

Comment:
```suggestion
/// assert_eq!(DisplayProtocolName(b"hello").to_string(), "hello");
```

The tests should probably use byte-strings. Plus a protocol-like string would be better IMO, something like `/tls/1.0.0`.

Reply:
It doesn’t really matter which bytes we use; I could make the test exhaustive by just using all 256 possible values.

Reply:
It doesn't but I think it is good to use test data that is reasonably close to the real world.